rust_src/src/fileio.rs

```diff
@@ -34,4 +36,20 @@ pub fn clear_buffer_auto_save_failure() {
     ThreadState::current_buffer().auto_save_failure_time = 0;
 }
 
+/// Return t if current buffer has been auto-saved recently.
+/// More precisely, if it has been auto-saved since last read from or saved
+/// in the visited file.  If the buffer has no visited file,
+/// then any auto-save counts as "recent".
+#[lisp_fn]
+pub fn recent_auto_save_p() -> LispObject {
```

Comment:
You could return bool here.

Reply:
True, that may make things clearer!

Reply:
You can write `cur_buf.modifications_since_save() < cur_buf.auto_save_modified` at the end of the function. `lisp_fn` converts the bool. Take a look at the coding book https://github.com/Wilfred/remacs/wiki/Porting-cookbook#return-types

Reply:
Ooof, stupid me, I'm not quite awake yet. :laughing:

Reply:
> I'm not quite awake yet.

Dito =)